skia/BUILD.gn

```diff
@@ -0,0 +1,832 @@
+      check_includes = false
+      configs = skia_library_configs
+
+      # "*" clobbers the current scope; append to existing configs
+      forward_variables_from(invoker,
+                             "*",
+                             [
+                               "configs",
+                               "public_defines",
+                               "sources_for_tests",
+                               "sources_when_disabled",
+                             ])
+      if (defined(invoker.configs)) {
+        configs += invoker.configs
+      }
+      all_dependent_configs = [ ":" + target_name + "_public" ]
+    }
+  } else {
+    skia_source_set(target_name) {
+      visibility = [ ":*" ]
+      configs = skia_library_configs
+
+      # "*" clobbers the current scope; append to existing configs
+      forward_variables_from(invoker,
+                             "*",
+                             [
+                               "configs",
+                               "public",
+                               "public_defines",
+                               "public_deps",
+                               "deps",
+                               "libs",
+                               "frameworks",
+                               "sources",
+                               "sources_for_tests",
+                               "sources_when_disabled",
+                             ])
+      if (defined(invoker.configs)) {
+        configs += invoker.configs
+      }
+      if (defined(invoker.sources_when_disabled)) {
+        sources = invoker.sources_when_disabled
+      }
+    }
+    if (defined(invoker.sources_for_tests)) {
+      skia_source_set(target_name + "_tests") {
+        visibility = [ ":*" ]
+      }
+    }
+  }
+}
+
+group("fontmgr_factory") {
+  public_deps = [ skia_fontmgr_factory ]
+}
+
+optional("fontmgr_empty_factory") {
+  enabled = true
+  sources = [ "$_skia_root/src/ports/SkFontMgr_empty_factory.cpp" ]
+}
+
+optional("fontmgr_android") {
+  enabled = skia_enable_fontmgr_android
+
+  deps = [
+    ":typeface_freetype",
+    "//third_party/expat",
+  ]
+  public = [ "$_skia_root/include/ports/SkFontMgr_android.h" ]
+  sources = [
+    "$_skia_root/src/ports/SkFontMgr_android.cpp",
+    "$_skia_root/src/ports/SkFontMgr_android_parser.cpp",
+    "$_skia_root/src/ports/SkFontMgr_android_parser.h",
+  ]
+}
+optional("fontmgr_android_factory") {
+  enabled = skia_enable_fontmgr_android
+  deps = [ ":fontmgr_android" ]
+  sources = [ "$_skia_root/src/ports/SkFontMgr_android_factory.cpp" ]
+}
+
+optional("fontmgr_custom") {
+  enabled =
+      skia_enable_fontmgr_custom_directory ||
+      skia_enable_fontmgr_custom_embedded || skia_enable_fontmgr_custom_empty
+
+  deps = [ ":typeface_freetype" ]
+  public = [ "$_skia_root/src/ports/SkFontMgr_custom.h" ]
+  sources = [ "$_skia_root/src/ports/SkFontMgr_custom.cpp" ]
+}
+
+optional("fontmgr_custom_directory") {
```

Comment:
You might be able to trim a few of these fontmgr configurations

Reply:
Yeah, good point. My plan is to do some deeper cleanups in a follow-up once this is landed and working.